hip-0000-improve-hedera-sharding.md

```diff
@@ -0,0 +1,175 @@
+
+------
+
+## Motivation
+
+Despite Hedera’s high throughput through the Gossip-about-Gossip protocol, the growing size of the network introduces challenges including:
+
+- Scalability bottlenecks from requiring each consensus node to maintain the full DAG’s recent state.
+- Increasing communication overhead from Gossip spreading every event to every node.
+- Security vulnerabilities stemming from static committee structures, making individual shards more susceptible to targeted attacks.
+
+To alleviate these limitations, this HIP introduces dynamic sharding with committee reconfiguration, supported by the modular integration of Block Nodes and unified Block Streams. These accommodate verification, block indexing, and shard state proofs efficiently—without overburdening consensus nodes.
+
+------
+
+## Rationale
+
+The proposed design extends research from systems such as OmniLedger and RapidChain while embracing Hedera-specific innovations like the Block Stream and Block Node models:
+
+- Reduces per-node workload by localizing data and limiting consensus responsibilities to relevant shards.
+- Parallelizes transaction processing across shards to increase throughput.
+- Incorporates Block Streams to unify state with event and transaction data, enabling downstream services (such as Block Nodes) to perform decentralized state verification and service provisioning.
+- Leverages mirror nodes as full archives for historical reference while Block Nodes supplement data discoverability and verification.
+
+Through this design, Hedera maintains consensus integrity, improves scalability, and supports decentralized application demands with dynamic committee realignment to reflect network conditions.
+
+------
+
+## User Stories
+
+1. Node Operator:
+   I want my node to only process shard-specific data and allow mirror or Block Nodes to manage full history and verification workloads.
+ 
+2. Application Developer:
+   I need reliable and low-latency cross-shard communication and access to trustworthy state proofs through Block Streams or Block Nodes.
+
+3. Data Service Provider:
+   I want to run a Block Node that ingests Block Streams to provide enhanced APIs and decentralized state proof services with optional monetization.
+ 
+4. Auditor / Security Engineer:
+   I require proofs of transaction execution and determinism through signed block data (TSS) and reproducibility via per-shard Block Streams.
+
+------
+
+## Specification
+
+### Architecture
+
+- Local Committees
+  - Composed of subsets of consensus nodes.
+  - Each manages its own shard’s ledger via a localized DAG.
+  - Each committee produces Block Streams (a unified record format) of their shard history.
+  - Uses shard-specific state proofs and aligns with the network-level TSS for consensus.
+
+- Global Committee
+  - Composed of rotating coordinators from Local Committees.
+  - Handles cross-shard transaction batching and finalizes inter-shard state updates.
```

Comment:
This introduces a bottleneck into the design. I had been thinking shards would communicate point to point rather than through a central communicator.